src/Shared/RedisUtility.cs

```diff
@@ -75,17 +75,25 @@ internal static object GetObjectFromBytes(byte[] dataAsBytes)
                 return null;
             }
 
-            BinaryFormatter binaryFormatter = new BinaryFormatter();
-            using (MemoryStream memoryStream = new MemoryStream(dataAsBytes, 0, dataAsBytes.Length))
+            try
             {
-                memoryStream.Seek(0, System.IO.SeekOrigin.Begin);
-                object retObject = (object)binaryFormatter.Deserialize(memoryStream);
-
-                if (retObject.GetType() == typeof(RedisNull))
+                BinaryFormatter binaryFormatter = new BinaryFormatter();
```

Comment:
Please add some unit tests to test this behaviour when exception occurs and when it doesn't occurs.